### ISSUE TITLE
Preserve additional information in extra file names/paths

### ISSUE THREAD

**Issue by azokthedefiler1** [Sonarr/Sonarr#5217]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

Sonarr will rename and move Jellyscrub created files in media folders which triggers Jellyscrub to recreate the original files. This has been observed with the Jellyscrub option "Save trickplay files within media folders" enabled. It is unknown if problem exists when that option is NOT checked.

![trickplay-media-folders](https://user-images.githubusercontent.com/61332872/199591415-7671c8e4-45d2-4250-9ee2-776f902969ea.png)

### Folder structure before Sonarr Media Management:
Series/
Series/Season 00/
Series/Season 00/Episode.mp4
Series/Season 00/Episode.nfo
Series/Season 00/Episode-thumb.jpg
Series/Season 00/trickplay/Episode-320.bif
Series/Season 00/trickplay/Episode-manifest.json

### Folder structure after Sonarr Media Management:
Series/
Series/Season 00/
Series/Season 00/Episode.mp4
Series/Season 00/Episode.nfo
Series/Season 00/Episode-thumb.jpg
Series/Season 00/Episode.bif
Series/Season 00/Episode.json

### What Sonarr did:
Sonarr strips **"-320"** off of the .bif file and moves it up into the Season folder.
Sonarr strips **"-manifest"** off of the .json file and moves it up into the Season folder.

### What Jellyscrub does after that:
Jellyscrub regenerates the now missing .bif and .json files.

### Describe the solution you'd like

Long term solution would be to handle the trickplay files correctly. Leave them in the **trickplay** folder and rename them preserving **-320** and **-manifest** in the filename.

Note that the **-320** in the BIF file is the resolution of the BIF file, which is the default preview resolution size (approx. 320x180). Matching against **-320.bif** would work for anyone using defaults, but matching against **-[0-9]{1,4}\.bif** should cover all cases.

### Describe alternatives you've considered

The easiest alternative I think would to be just ignore the **trickplay** folder during media management runs.

Another alternative is to add an option to Sonarr in the **Settings / Media Management** section to exclude folders/files by extension or glob.

### Anything else?

Here's a link to Jellyscrub so you can check it out:

[Jellyscrub](https://github.com/nicknsy/jellyscrub)

**Comment by edrock200** [comment]
I can confirm this is also the case for sideloaded bifs from Emby as well.

**Comment by marioK93** [comment]
I have been putting off a huge rename of my library for a while because of this issue.

I discovered yesterday that you can run custom scripts in sonarr on certain triggers: https://wiki.servarr.com/sonarr/custom-scripts

I put together a quick script for my library that handles moving the trickplay files and also correctly moves the image from the "metadata" folder jellyfin creates into the new directory as this wasn't handled correctly by sonarr.

It should work for most libraries no problem but i recommend testing it first of course. I manually checked the directories for most of my library to make sure things worked and some cases still needed some manual intervention, so maybe others can help improve the script.



```shell
#!/usr/bin/env bash

#get new and old episode relative paths from sonarr
IFS='|' read -r -a new_relative_paths <<< "$sonarr_episodefile_relativepaths"
IFS='|' read -r -a previous_relative_paths <<< "$sonarr_episodefile_previouspaths"

#loop counter for second array in loop below
i=0

#loop through new paths
for path in "${new_relative_paths[@]}"
do
	#Sort variables for new episodes, full path, directory and name without extension
	full__new_episode_path="$sonarr_series_path/$path"
	new_episode_dir=$(dirname "$full__new_episode_path")
	new_episode_file_path=$(basename  "$full__new_episode_path")
	new_episode_file_name="${new_episode_file_path%.*}"

	#Sort variables for old episodes, full path, directory and name without extension, counter above used here
	full__old_episode_path="${previous_relative_paths[$i]}"
	old_episode_dir=$(dirname "$full__old_episode_path")
	old_episode_file_path=$(basename  "$full__old_episode_path")
	old_episode_file_name="${old_episode_file_path%.*}"
	
	#check if new trickplay directory exists and create if not including .ignore file
	new_trickplay_dir="$new_episode_dir/trickplay"
	if [ ! -d "$new_trickplay_dir" ];
	then
		mkdir "$new_trickplay_dir"
		touch "$new_trickplay_dir/.ignore"
	fi
	
	#different cases for different possible file locations based on what i found in my own library
	
	#checks if files have already been incorrectly moved to new episode directory by sonarr
	moved_bif_file="$new_episode_dir/$new_episode_file_name.bif"
	moved_json_file="$new_episode_dir/$new_episode_file_name.json"
	if [ -f "$moved_bif_file" ];
	then
		echo "Moved Already Bif"
		new_bif_file="$new_trickplay_dir/$new_episode_file_name-320.bif"
		mv "$moved_bif_file" "$new_bif_file"
	fi
	if [ -f "$moved_json_file" ];
	then
		echo "Moved Already Json"
		new_json_file="$new_trickplay_dir/$new_episode_file_name-manifest.json"
		mv "$moved_json_file" "$new_json_file"
	fi
	
	#checks if files are in old trickplay directory
	old_bif_trickplay_file="$old_episode_dir/trickplay/$old_episode_file_name-320.bif"
	old_json_trickplay_file="$old_episode_dir/trickplay/$old_episode_file_name-manifest.json"
	if [ -f "$old_bif_trickplay_file" ];
	then
		echo "Old Trickplay Bif"
		new_bif_file="$new_trickplay_dir/$new_episode_file_name-320.bif"
		mv "$old_bif_trickplay_file" "$new_bif_file"
	fi
	if [ -f "$old_json_trickplay_file" ];
	then
		echo "Old Trickplay Json"
		new_json_file="$new_trickplay_dir/$new_episode_file_name-manifest.json"
		mv "$old_json_trickplay_file" "$new_json_file"
	fi	


	#checks if files are in old episode directory
	old_bif_file="$old_episode_dir/$old_episode_file_name.bif"
	old_json_file="$old_episode_dir/$old_episode_file_name.json"	
	if [ -f "$old_bif_file" ];
	then
		echo "Old Episode Bif"
		new_bif_file="$new_trickplay_dir/$new_episode_file_name-320.bif"
		mv "$old_bif_file" "$new_bif_file"
	fi
	if [ -f "$old_json_file" ];
	then
		echo "Old Episode json"
		new_json_file="$new_trickplay_dir/$new_episode_file_name-manifest.json"
		mv "$old_json_file" "$new_json_file"
	fi

	#remove old trickplay directory
	old_trickplay_dir="$old_episode_dir/trickplay"
	if [ -d "$old_trickplay_dir" ];
	then
		rm -rf "$old_trickplay_dir"
	fi
	
	
	
	
	#handle moving images from previous episode metadata folder as sonarr wouldn't handle this correctly
	
	#create folder if doesn't already exist
	new_metadata_dir="$new_episode_dir/metadata"
	if [ ! -d "$new_metadata_dir" ];
	then
		mkdir "$new_metadata_dir"	
	fi
	
	
	#check for image in old metadata folder
	old_metadata_image_file="$old_episode_dir/metadata/$old_episode_file_name.jpg"
	if [ -f "$old_metadata_image_file" ];
	then
		echo "Old Episode Image"
		new_image_file="$new_episode_dir/metadata/$new_episode_file_name.jpg"
		mv "$old_metadata_image_file" "$new_image_file"
	fi
	
	
	#check for image in old episode directory
	old_image_file="$old_episode_dir/$old_episode_file_name.jpg"
	if [ -f "$old_image_file" ];
	then
		echo "Old Episode Image"
		new_image_file="$new_episode_dir/metadata/$new_episode_file_name.jpg"
		mv "$old_image_file" "$new_image_file"
	fi
	
	
	#check for image in new episode directory
	moved_image_file="$new_episode_dir/$old_episode_file_name.jpg"
	if [ -f "$moved_image_file" ];
	then
		echo "Moved Episode Image"
		new_image_file="$new_episode_dir/metadata/$new_episode_file_name.jpg"
		mv "$moved_image_file" "$new_image_file"
	fi


	#remove old metadata folder
	old_metadata_dir="$old_episode_dir/metadata"
	if [ -d "$old_metadata_dir" ];
	then
		rm -rf "$old_metadata_dir"
	fi
	
	
	i=$[i + 1]
done

**Comment by zentaos** [comment]
This is a present issue for Emby too.
As soon as Emby detect the new files it creates the .bif files.

But when I scan & rename episodes in Sonarr, it wont move the .bif files

**Comment by kingy444** [comment]
Adding a note that the issue in Emby needs to also account for `-320-x.bif` where the `x` is the number of seconds the bif file is generated for - so you could have things like `320-5.bif`,`320-10.bif` and `320-20.bif` dependent on user settings

**Comment by astubbs** [comment]
Why don’t we make sonarr rename the prefix part of every file and directory found in the same dir, that has the same prefix?

**Comment by markus101** [comment]
Sonarr only renames files it's tracking, this avoids rescanning multiple times for every file renamed to see if there might be something to rename and would allow Sonarr to display which files are present in the UI at some point.

The correct solution here would be to track these files and rename them the same way others are, a naive approach that renames files/folders with the same prefix is better suited for a Custom Script.

**Comment by kingy444** [comment]
I did notice a bug recently but it was minor and didn't think much of it until the above comments

Essentially I renamed 1 file from a season, but every single extra file (including those for episodes I didn't request a rename on) was renamed 

I didn't care much because I planned to rename them all after the fact (testing new naming standard)

But it does indicate some files other than just those requested are being touched, perhaps extending to match all files would suffice

For instance the issue I face currently is my EMBY generated bif files are `<episodename>-320-10.bif` and while a SONARR rename respects a `-thumb.jpg` rename when it is included metadata the the `-320-10` is dropped from the bif file

**Comment by astubbs** [comment]
> The correct solution here would be to track these files and rename them the same way others are

Got it. Would you be open to accepting a PR that does this?

**Comment by kingy444** [comment]
TLdR would be:
My bif files are already being monitored but their suffix is not respected during a rename

**Comment by markus101** [comment]
They're likely getting picked up under something else that doesn't understand the more complex naming required to rename them properly.

Yes, we'd be open to a PR to support tracking and renaming these files properly.

**Comment by markus101** [comment]
This also extends to other supporting files that can contain additional information in the file names, such as mp3 files with language as explained in https://github.com/Sonarr/Sonarr/issues/7535

Perhaps the suggested solution of handling renaming and preserving by the extra information in the file name is sufficient to cover all these cases, but that information should be extracted and stored in the DB to avoid having to re-extract it from the file name.
